homeassistant/components/aranet4/config_flow.py

```diff
@@ -0,0 +1,129 @@
+"""Config flow for Aranet4 integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from aranet4.client import Aranet4Advertisement, Version as AranetVersion
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfoBleak,
+    async_discovered_service_info,
+)
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_VERSION = AranetVersion(1, 2, 0)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Aranet4."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Set up a new config flow for Aranet4."""
+        self._discovery_info: BluetoothServiceInfoBleak | None = None
+        self._discovered_device: Aranet4Advertisement | None = None
+        self._discovered_devices: dict[str, tuple[str, Aranet4Advertisement]] = {}
+
+    def _errors_from_adv(self, adv: Aranet4Advertisement) -> FlowResult | None:
+        """Get any configuration errors that apply to an advertisement."""
+        # Old versions of firmware don't expose sensor data in advertisements.
+        if not adv.manufacturer_data or adv.manufacturer_data.version < MIN_VERSION:
+            return self.async_abort(reason="outdated_version")
+
+        # If integrations are disabled, we get no sensor data.
+        if not adv.manufacturer_data.integrations:
+            return self.async_abort(reason="integrations_disabled")
+
+        return None
+
+    async def async_step_bluetooth(
+        self, discovery_info: BluetoothServiceInfoBleak
+    ) -> FlowResult:
+        """Handle the Bluetooth discovery step."""
+        await self.async_set_unique_id(discovery_info.address)
+        self._abort_if_unique_id_configured()
+        adv = Aranet4Advertisement(discovery_info.device, discovery_info.advertisement)
+        errors = self._errors_from_adv(adv)
+        if errors:
+            return errors
```

Comment:
```suggestion
        self._errors_from_adv(adv)
```

Probably rename this to `_raise_for_advertisement_errors`